lib/domain/usecases/categories/categories_by_filter_result.dart

```diff
@@ -0,0 +1,18 @@
+
+
+import 'package:openflutterecommerce/data/abstract/model/category.dart';
+
+class CategoriesByFilterResult {
+  final List<Category> categories;
+  final int quantity;
+  final Exception exception;
+
+  CategoriesByFilterResult(
+    this.categories,
+    this.quantity,{
+      this.exception
+    }
+  );
+  
+  bool get validResults => exception == null;
```

Comment:
it is better to name boolean properties starting from is or has. In this exact case: isResultValid. Source: "Clean Code"

Reply:
Good point. Thanks.